core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JvmStartupOptimizerArchiveBuildStep.java

```diff
@@ -117,12 +117,12 @@ public void build(Optional<AppCDSRequestedBuildItem> requested,
             }
         }
 
-        appCDS.produce(new AppCDSResultBuildItem(archivePath));
+        appCDS.produce(new JvmStartupOptimizerArchiveResultBuildItem(archivePath));
         artifactResult.produce(new ArtifactResultBuildItem(archivePath, "appCDS", Collections.emptyMap()));
```

Comment:
This method looks generic but we still reference `appCDS` here.

Reply:
I'll keep the `"appCDS"` artifact result type  for now, but that also might need to be changed in the future